### ISSUE TITLE
Optimize memory usage in `_initialize_alpha` by removing full matrix construction and optimizing matrix operations.

### ISSUE THREAD

**Issue by Bucanero06** [PyDMD/PyDMD#539]
This change request modifies the `_initialize_alpha` function in the BOPDMD implementation to enhance memory efficiency. The original implementation created a large diagonal matrix `T`, which led to excessive memory consumption for large datasets.

**Key Changes**:
1. **Avoid Full Matrix Construction**: Replaced the construction of the full diagonal matrix `T` with direct element-wise operations. This approach significantly reduces memory usage.
2. **Optimized Matrix Operations**: Ensured that matrix multiplications are efficiently managed to avoid redundant computations.

With these changes, I can handle larger datasets without encountering memory allocation errors.

Please review the changes for potential integration and feedback from the community.

```
def _initialize_alpha(self):
    """
    Uses projected trapezoidal rule to approximate the eigenvalues of A in
        z' = Az.
    The computed eigenvalues will serve as our initial guess for alpha.

    :return: Approximated eigenvalues of the matrix A.
    :rtype: numpy.ndarray
    """
    # Project the snapshot data onto the projection basis.
    ux = self._proj_basis.conj().T.dot(self.snapshots)
    ux1 = ux[:, :-1]
    ux2 = ux[:, 1:]

-   # Define the diagonal matrix T as the following.
-   t1 = self._time[:-1]
-   t2 = self._time[1:]
-   T = np.diag(t2 - t1)
-
-   # Define the matrices Y and Z as the following and compute the
-   # rank-truncated SVD of Y.
-   Y = (ux1 + ux2) / 2
-   Z = (ux2 - ux1).dot(np.linalg.inv(T))
+   # Define the time differences directly without forming a large matrix.
+   t_diffs = self._time[1:] - self._time[:-1]
+
+   # Define the matrices Y and Z as the following.
+   Y = (ux1 + ux2) / 2
+   Z = (ux2 - ux1) / t_diffs  # Element-wise division by time differences.

    # Compute the rank-truncated SVD of Y.
    U, s, V = compute_svd(Y, self._svd_rank)
    S = np.diag(s)

-   # Compute the matrix Atilde and return its eigenvalues.
-   Atilde = np.linalg.multi_dot([U.conj().T, Z, V, np.linalg.inv(S)])
+   # Compute the matrix Atilde using efficient matrix products and return its eigenvalues.
+   Atilde = U.conj().T @ Z @ V @ np.linalg.inv(S)

    return np.linalg.eig(Atilde)[0]
```

**Comment by mtezzele** [comment]
Dear @Bucanero06 , thank you so much for this enhancement. I would keep the `multi_dot` since it is the optimized version of the `dot` operators. Please go ahead and open a PR with this modifications and mention this issue so everything is linked together.

**Comment by Bucanero06** [comment]
Sounds good, I left multi_dot untouched and only changed  line `Z = (ux2 - ux1) / (self._time[1:] - self._time[:-1])`; PR #540

**Comment by fandreuz** [comment]
Fixed by #540
